src/Microsoft.Management.Configuration/Microsoft.Management.Configuration.idl

```diff
@@ -0,0 +1,605 @@
+        ConfigurationUnit();
+
+        // The name of the unit being configured; not a name for this instance.
+        // Can be changed if a mutable unit.
+        String UnitName;
+
+        // An identifier used to uniquely identify the instance of a configuration unit on the system.
+        // Immutable, although the unit can be removed and an identical unit applied later with a different identifier.
+        Guid InstanceIdentifier{ get; };
+
+        // The identifier name of this instance within the set.
+        // Can be changed if a mutable unit.
+        String Identifier;
+
+        // Describes how this configuration unit will be used.
+        // Can be changed if a mutable unit.
+        ConfigurationUnitIntent Intent;
+
+        // The configuration units that this unit depends on.
+        // Can be set on a mutable configuration unit; this will copy the items into the internal storage.
+        Windows.Foundation.Collections.IVectorView<String> Dependencies;
+
+        // Contains the values that are for use by the configuration system, related to this unit.
+        Windows.Foundation.Collections.ValueSet Directives{ get; };
+
+        // Contains the values that are for use by the configuration unit itself.
+        Windows.Foundation.Collections.ValueSet Settings{ get; };
+
+        // Contains information on the origin of the configuration unit.
+        // May be null if ConfigurationProcessor.GetDetailsAsync has not been called yet.
+        IConfigurationUnitProcessorDetails Details{ get; };
+
+        // The current state of the configuration unit.
+        ConfigurationUnitState State{ get; };
+
+        // Contains information on the result of the latest attempt to apply the configuration unit.
+        ConfigurationUnitResultInformation ResultInformation{ get; };
+
+        // Allows for control over whether this unit should be applied when the set containing it is applied.
+        Boolean ShouldApply;
+    }
+
+    // The change event type that has occurred for a configuration set change.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    enum ConfigurationSetChangeEventType
+    {
+        Unknown,
+        SetStateChanged,
+        UnitStateChanged,
+    };
+        
+    // The change data sent about changes to a specific set.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass ConfigurationSetChangeData
+    {
+        // The change event type that occurred.
+        ConfigurationSetChangeEventType Change{ get; };
+
+        // The state of the configuration set for this event (the ConfigurationSet can be used to get the current state, which may be different).
+        ConfigurationSetState SetState{ get; };
+            
+        // The state of the configuration unit for this event (the ConfigurationUnit can be used to get the current state, which may be different).
+        ConfigurationUnitState UnitState{ get; };

```

Comment:
We only have one of those at a time depending on the type?

Reply:
Yes, and it doesn't seem worthwhile to create three runtime classes and require a QI to get any useful data out.